js/util/__tests__/extractProps.spec.ts

```diff
@@ -71,6 +72,18 @@ describe('Extracting Props with a configuration', () => {
     );
   });
 
+  test('arrays flatten properly', () => {
+    const obj2 = _.merge({}, obj1, {
+      props: {
+        a: [3, 4, 5],
+      },
+    });
+
+    expect(extractProps('Element', obj2, config)).toEqual(
+      '[a.0=3];[a.1=4];[a.2=5];[c=true];'
```

Comment:
I'm not entirely sure this is actually correct.  Since heap doesn't really support array props, this is effectively meaningless unless index zero maps to one thing, index one to something else, etc.

This is fine for now, but we should discuss this.